js/block.js

```diff
@@ -639,6 +718,22 @@ var convert = {
     any: function(text){return util.isNumber(text) ? +text : text; }
 };
 
+/*****************
+*
+*  wb-contains
+*
+*  Instantiated as new WBContains or as <wb-contains>. Contains is a semi-block. It never
+*  exists by itself, only as a child of a <wb-context> (and every <wb-context has at least one
+*  <wb-contains> child, implicitly if not explicitly). Sometimes we need to test for other blocks
+*  being in a <wb-context>, which is why sometimes it is block-ish.
+*
```

Comment:
The fact that we test for containment in `wb-contains` rather than `wb-context` may be a bug, but I don't know how else to handle `if-then` blocks.